WooCommerce/Classes/ViewRelated/Themes/ThemeInstaller.swift

```diff
@@ -0,0 +1,128 @@
+import Yosemite
+import enum Networking.InstallThemeError
+
+protocol ThemeInstallerProtocol {
+    func install(themeID: String, siteID: Int64) async throws
+
+    func scheduleThemeInstall(themeID: String, siteID: Int64)
+
+    func installPendingTheme(siteID: Int64) async throws
+}
+
+/// Helper to install and activate theme
+///
+struct DefaultThemeInstaller: ThemeInstallerProtocol {
+    private let userDefaults: UserDefaults
+    private let stores: StoresManager
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         userDefaults: UserDefaults = .standard) {
+        self.stores = stores
+        self.userDefaults = userDefaults
+    }
+
+    /// Installs and activates the theme
+    /// - Parameters:
+    ///   - themeID: ID of the theme to be installed and activated
+    ///   - siteID: site ID to install and activate the theme
+    func install(themeID: String, siteID: Int64) async throws {
+        try await installAndActivateTheme(themeID: themeID, siteID: siteID)
+    }
+
+    /// Schedules the given themeID for installation and activation
+    /// - Parameters:
+    ///   - themeID: ID of the theme to be installed and activated
+    ///   - siteID: site ID to install and activate the theme
+    func scheduleThemeInstall(themeID: String, siteID: Int64) {
+        userDefaults.setThemeID(themeID: themeID, for: siteID)
+    }
+
+    /// Installs any pending theme for the given site ID
+    /// - Parameter siteID: site ID to install and activate the theme
+    func installPendingTheme(siteID: Int64) async throws {
+        guard let themeID = userDefaults.themeID(for: siteID) else {
+            return DDLogInfo("No pending theme installation.")
+        }
+
+        DDLogInfo("Attempt to perform pending theme installation. Theme ID: \(themeID), Site ID : \(siteID)")
+        try await installAndActivateTheme(themeID: themeID, siteID: siteID)
+        userDefaults.removeThemeID(for: siteID)
+    }
+}
+
+private extension DefaultThemeInstaller {
+    func installAndActivateTheme(themeID: String, siteID: Int64) async throws {
+        try await installTheme(themeID: themeID, siteID: siteID)
+        try await activateTheme(themeID: themeID, siteID: siteID)
+    }
+
+    @MainActor
+    func installTheme(themeID: String,
+                      siteID: Int64) async throws {
+        try await withCheckedThrowingContinuation { continuation in
+            stores.dispatch(WordPressThemeAction.installTheme(themeID: themeID, siteID: siteID) { result in
+                switch result {
+                case .success:
+                    continuation.resume()
+                case .failure(let error):
+                    if error as? InstallThemeError == .themeAlreadyInstalled {
+                        DDLogInfo("Theme already installed.")
+                        continuation.resume()
+                    } else {
+                        DDLogError("Theme installation failed: \(error)")
```

Comment:
Nit: We should add the ⛔️ emoji here to make it easier to notice the error in the log.

Reply:
Done in f44b287a43fffa8e521db686d38d225c58204c41